phenopacketlab-frontend/src/app/component/phenopacket/phenopacket.component.ts

```diff
@@ -46,16 +47,13 @@ export class PhenopacketComponent implements OnInit, OnDestroy {
 
   ngOnInit(): void {
     // retrieve phenopacket to edit
-    this.phenopacketListSubscription = this.phenopacketService.getPhenopacketList().subscribe(list => {
-      if (list) {
-        for (const pheno of list) {
-          if (pheno.id === this.phenopacketId) {
-            // deep copy of object so we do not modify by reference
-            this.phenopacket = Utils.clone(pheno);
-          }
-        }
-      }
-    });
+    this.phenopacketListSubscription = this.phenopacketService.getPhenopacketList()
+      .pipe(
+        map(phenopackets => phenopackets.find(pheno => pheno.id === this.phenopacketId)))
+      .subscribe(phenopacket => {
+        // deep copy of object so we do not modify by reference
+        this.phenopacket = Utils.clone(phenopacket);
```

Comment:
This is nullable now. A condition that should never be met maybe we should do something if it happens like "console.error" or the like.